src/main/java/teammates/ui/webapi/action/SearchSessionsAction.java

```diff
@@ -0,0 +1,142 @@
+package teammates.ui.webapi.action;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+import teammates.common.util.StringHelper;
+import teammates.ui.webapi.output.SearchSessionsData;
+import teammates.ui.webapi.output.StudentSessionsData;
+
+/**
+ * Searches for sessions.
+ */
+public class SearchSessionsAction extends Action {
+
+    private static final String OPEN_CLOSE_DATES_SESSION_TEMPLATE = "[%s - %s]";
+
+    private Set<String> courseIds = new HashSet<>();
+    private Map<String, List<FeedbackSessionAttributes>> courseIdToOpenFeedbackSessionsMap = new HashMap<>();
+    private Map<String, List<FeedbackSessionAttributes>> courseIdToNotOpenFeedbackSessionsMap = new HashMap<>();
+    private Map<String, List<FeedbackSessionAttributes>> courseIdToPublishedFeedbackSessionsMap = new HashMap<>();
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // Only admins can search for sessions
+        if (!userInfo.isAdmin) {
+            throw new UnauthorizedAccessException("Admin privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        String searchKey = getNonNullRequestParamValue(Const.ParamsNames.ADMIN_SEARCH_KEY);
+        List<StudentAttributes> students = logic.searchStudentsInWholeSystem(searchKey).studentList;
```

Comment:
Is this endpoint also called for instructor search? if so, we would also have to search for instructors and maybe return a {students: <link for students> and instructors: <links for instructors object>}

Reply:
As far as I can see, session links are only returned for the admin search: https://github.com/dvrylc/teammates/pull/1#issuecomment-589702428